Functions/Functions/User/GetUserFunction.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using OidcApiAuthorization.Abstractions;
+using OidcApiAuthorization.Models;
+using System.Linq;
+
+namespace SSW.Rules.Functions
+{
+    public class GetUserFunction
+    {
+        private readonly RulesDbContext _dbContext;
+        private readonly IApiAuthorization _apiAuthorization;
+
+        public GetUserFunction(RulesDbContext dbContext, IApiAuthorization apiAuthorization)
+        {
+            _dbContext = dbContext;
+            _apiAuthorization = apiAuthorization;
+        }
+
+        [FunctionName("GetUserFunction")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            ApiAuthorizationResult authorizationResult = await _apiAuthorization.AuthorizeAsync(req.Headers);
+
+            if (authorizationResult.Failed)
+            {
+                log.LogWarning(authorizationResult.FailureReason);
+                return new UnauthorizedResult();
+            }
+
+            log.LogWarning($"HTTP trigger function {nameof(GetUserFunction)} request is authorized.");
+
+            string UserId = req.Query["user_id"];
+
+            if (string.IsNullOrEmpty(UserId))
+            {
+                return new BadRequestObjectResult(new
+                {
+                    message = "Missing or empty user_id param",
+                });
+            }
+
+            var result = await _dbContext.Users.Query(q => q.Where(w => w.UserId == UserId));
+            if (result.Count() == 0)
+            {
+                log.LogInformation($"Could not find results for user: {UserId}");
+                return new BadRequestObjectResult(new
+                {
+                    message = "User " + UserId + " was not found",
+                });
+            }
+
+            User user = result.FirstOrDefault();
+
+            if (string.IsNullOrEmpty(user?.CommentsUserId))
+            {
+                return new OkObjectResult(new
+                {
+                    user = user,
+                    commentsConnected = false,
+                });
+            }
+

```

Comment:
I would make the second return part of an else statement but you could argue this is a stylistic decision. I prefer it because when someone goes to edit it later the structure is maintained if they add things in and it is more difficult for the logic flow to be "lost" through future maintenance.

Also another thing you could do is simply return the object and set the comments connected boolean according to the result of string.isnullorempty, this would be more compact than the current code. I would likely opt for this method unless I thought that it was likely the result would change in the future to be more different. I like this method because both results are valid, but they return different information.